### PR TITLE
issue: 1117626 Move epoll poll_os logic to the internal thread

### PR DESCRIPTION
**Pull request by liranoz12**

### CODE REVIEW

**Comment by liranoz12** [comment]
The commit is based on #445

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1634/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1642/ for details (Mellanox internal link).

**Comment by DanielLibenson** [comment]
bot:reset

**Comment by liranoz12** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1643/ for details (Mellanox internal link).

File: src/vma/iomux/io_mux_call.cpp [hunk 1]
```diff
@@ -245,37 +245,54 @@ void io_mux_call::check_offloaded_rsockets(uint64_t *p_poll_sn)
...
 
+bool io_mux_call::handle_os_countdown(int &poll_os_countdown)
+{
+	/*
+	 * Poll OS when count down reaches zero. This honors CQ-OS ratio.
+	 * This also handles the 0 ratio case - do not poll OS at all.
+	 */
+	if (poll_os_countdown-- == 0 && m_n_sysvar_select_poll_os_ratio > 0) {
+		bool cq_ready = wait_os(true);
+		if (cq_ready) {
```
Comment (DanielLibenson) [inline src/vma/iomux/io_mux_call.cpp]: Please consider to move wait_os() inside if statement.

Comment (liranoz12) [reply src/vma/iomux/io_mux_call.cpp]: :+1:

**Comment by DanielLibenson** [comment]
:+1:

**Comment by liranoz12** [comment]
bot:retest

**Comment by liranoz12** [comment]
Merged #460

**Pull request closed without merging**